nested.py

```diff
@@ -7,16 +7,45 @@
 
 import sys
 
-
 def is_nested(line):
-    """Validate a single input line for correct nesting"""
-    pass
+    brackets = {
+        "(":["open"], ")":["close", "("], 
+        "[":["open"], "]":["close", "["], 
+        "{":["open"], "}":["close", "{"], 
+        "<":["open"], ">":["close", "<"], 
+    }
+    
+    stack = []
+    tokens = 0
+    
+    for i, c in enumerate(line, start=0):
+        if line[i:i+2] == "(*":
+            stack.append("(*")
+        elif line[i-1:i+1] == "*)" and line[i-2] != "(":
+            if stack[-1] == "(*":stack.pop()
+            else: return "NO {}".format(tokens)
+        elif c in brackets:
+            tokens += 1
+            if brackets[c][0] == "open": stack.append(c)
+            elif stack[-1] == brackets[c][1]: stack.pop()
+            else: return "NO {}".format(tokens)
+        else: tokens += 1
```

Comment:
Boo hiss! Always a new line afer `:` in python.
In this case it is super clean tho.

I also just realized we're not in a while loop.
That's awesome because this way, I'd have to think, is a little more difficult.